### PR TITLE
Minor flag refactoring

### PR DESCRIPTION
**Pull request by ifimfree**
Per Issue #5173, added enums to represent flags in 4 files.

### CODE REVIEW

**Comment by tupaschoal** [comment]
Thanks for submitting your changes! We're on code freeze for v0.2.6, just taking bug fixes, so please bear with us as it might take us longer to review :)

File: src/openrct2/EditorObjectSelectionSession.h [hunk 1]
```diff
@@ -14,6 +14,15 @@
 
 #include <vector>
 
+enum INPUT_FLAGS
+{
+    INPUT_FLAG_EDITOR_OBJECT_1 = (1 << 1),
```
Comment (Gymnasiast) [inline src/openrct2/EditorObjectSelectionSession.h:19]: You can remove the blank lines between the enum members.

File: src/openrct2/EditorObjectSelectionSession.h [hunk 2]
```diff
@@ -14,6 +14,15 @@
...
+
+    INPUT_FLAG_EDITOR_OBJECT_2 = (1 << 2),
+
+    INPUT_FLAG_EDITOR_OBJECT_3 = (1 << 3)
```
Comment (Gymnasiast) [inline src/openrct2/EditorObjectSelectionSession.h:23]:
```suggestion
    INPUT_FLAG_EDITOR_OBJECT_ALWAYS_REQUIRED = (1 << 3)
```

File: src/openrct2/ride/Ride.h [hunk 3]
```diff
@@ -950,13 +950,28 @@ enum
...
     RIDE_MODIFY_DEMOLISH,
-    RIDE_MODIFY_RENEW
+    RIDE_MODIFY_RENEW,
 };
 
 enum
 {
     RIDE_ISSUE_NONE = 0,
-    RIDE_ISSUE_GUESTS_STUCK = (1 << 0),
+    RIDE_ISSUE_GUESTS_STUCK = (1 << 0)
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.h:959]: Revert. Trailing commas are nice.

File: src/openrct2/world/Footpath.cpp [hunk 4]
```diff
@@ -1275,7 +1275,7 @@ static bool get_next_direction(int32_t edges, int32_t* direction)
  *
  *  rct2: 0x0069AC1A
  * @param flags (1 << 0): Ignore queues
- *              (1 << 5): Unown
+ *              (1 << 5): Unkown
```
Comment (Gymnasiast) [inline src/openrct2/world/Footpath.cpp:1278]: Revert. "Unown" is proper English ("to deprive of ownership").

File: src/openrct2/world/Footpath.cpp [hunk 5]
```diff
@@ -1315,21 +1315,21 @@ static int32_t footpath_is_connected_to_map_edge_recurse(
...
+        if (!(flags & FOOTPATH_CONNECTED_MAP_EDGE_IGNORE_QUEUES))
         {
             if (tileElement->AsPath()->IsQueue())
             {
                 continue;
             }
         }
 
-        if (flags & (1 << 5))
+        if (flags & FOOTPATH_CONNECTED_MAP_EDGE_UNKNOWN)
```
Comment (Gymnasiast) [inline src/openrct2/world/Footpath.cpp:1326]: Perhaps checking the callers of this function will reveal what this flag is for. You can then also replace this flag in the callers of this function.

**Pull request merged**